ec2utils/ec2deprecateimg/lib/ec2utils/ec2deprecateimg.py

```diff
@@ -0,0 +1,401 @@
+                    msg += 'deprecation search. Image ID: %s' %img.id
+                    print msg
+                continue
+            if imgName == img.name:
+                images.append(img)
+
+        return images
+
+    #----------------------------------------------------------------------
+    def _findImagesByNameFragment(self, nameFragment, filterReplImg=None):
+        """Find images by string matching of the fragment with the name"""
+        images = []
+        myImages = self._getAllTypeMatchImages(filterReplImg)
+        for img in myImages:
+            if not img.name:
+                if filterReplImg:
+                    msg = 'WARNING: Found image with no name, ignoring for '
+                    msg += 'deprecation search. Image ID: %s' %img.id
+                    print msg
+                continue
+            if img.name.find(nameFragment) != -1:
+                images.append(img)
+
+        return images
+
+    #----------------------------------------------------------------------
+    def _findImagesByNameRegexMatch(self, expression, filterReplImg=None):
+        """Find images by match the name with the given regular expression"""
+        images = []
+        myImages = self._getAllTypeMatchImages(filterReplImg)
+        match = re.compile(expression)
+        for img in myImages:
+            if not img.name:
+                if filterReplImg:
+                    msg = 'WARNING: Found image with no name, ignoring for '
+                    msg += 'deprecation search. Image ID: %s' %img.id
+                    print msg
+                continue
+            if match.match(img.name):
+                images.append(img)
+
+        return images
+
+    #----------------------------------------------------------------------
+    def _formatDate(self, date):
+        """Format the date to YYYYMMDD"""
+        year = date.year
+        month = date.month
+        day = date.day
+        date = '%s' % year
+        for item in [month, day]:
+            if item > 9:
+                date += '%s' %item
+            else:
+                date += '0%s' %item
+
+        return date
+
+    #----------------------------------------------------------------------
+    def _getAccessKeys(self):
+        """Get the access keys for the account from the configuration"""
+	if not self.config.has_option(self.account, 'access_key_id'):
+	    msg = 'No access_key_id option found for %s' %self.account
+	    raise EC2DeprecateImgException(msg)
+	if not self.config.has_option(self.account, 'secret_access_key'):
+	    msg = 'No secret_access_key option found for %s' %self.acount
+	    raise EC2DeprecateImgException(msg)
+
+	accessKey = self.config.get(self.account, 'access_key_id')
+	secretKey = self.config.get(self.account, 'secret_access_key')
+
+        return accessKey, secretKey
+        
+    #----------------------------------------------------------------------
+    def _getConfig(self):
```

Comment:
I agree. However as the class needs to hold the configFile name anyway, the name is needed in a message in _verifyAccount() I chose to use a member rather than passing the configFilePath around to multiple methods.